### PR TITLE
ui: dmoji

### PR DESCRIPTION
**Pull request by ZwX1616**

### CODE REVIEW

File: selfdrive/ui/ui.h [hunk 1]
```diff
@@ -25,6 +25,17 @@ typedef cereal::CarControl::HUDControl::AudibleAlert AudibleAlert;
 
 const mat3 DEFAULT_CALIBRATION = {{ 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0 }};
 
+const int FACE_KPTS_SIZE = 31;
```
Comment (adeebshihadeh) [inline selfdrive/ui/ui.h:28]: this shouldn't be hardcoded

File: selfdrive/ui/ui.cc [hunk 2]
```diff
@@ -226,6 +259,12 @@ UIState::UIState(QObject *parent) : QObject(parent) {
   timer = new QTimer(this);
   QObject::connect(timer, &QTimer::timeout, this, &UIState::update);
   timer->start(1000 / UI_FREQ);
+
+  // initial dm draw
```
Comment (adeebshihadeh) [inline selfdrive/ui/ui.cc:263]: what's this for? don't we hide the icon until the DM packet is seen?

File: selfdrive/ui/qt/onroad.cc [hunk 3]
```diff
@@ -273,9 +273,11 @@ void AnnotatedCameraWidget::updateState(const UIState &s) {
   setProperty("is_cruise_set", cruise_set);
   setProperty("is_metric", s.scene.is_metric);
   setProperty("speed", cur_speed);
+  setProperty("isStandstill", sm["carState"].getCarState().getStandstill());
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.cc:276]: unused?

File: selfdrive/ui/qt/onroad.cc [hunk 4]
```diff
@@ -273,9 +273,11 @@ void AnnotatedCameraWidget::updateState(const UIState &s) {
...
   setProperty("setSpeed", set_speed);
   setProperty("speedUnit", s.scene.is_metric ? tr("km/h") : tr("mph"));
   setProperty("hideDM", cs.getAlertSize() != cereal::ControlsState::AlertSize::NONE);
+  setProperty("dm_fade_state", fmax(0.0, fmin(1.0, dm_fade_state+0.2*(0.5-(float)(dmActive)))));
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.cc:280]: why use a property here? seems unnecessary

File: selfdrive/ui/ui.cc (hunk 2 above)
Comment (ZwX1616) [reply selfdrive/ui/ui.cc:263]: no? we only hide dm on alerts

File: selfdrive/ui/qt/onroad.cc (hunk 3 above)
Comment (ZwX1616) [reply selfdrive/ui/qt/onroad.cc:276]: you're right. fixing

File: selfdrive/ui/qt/onroad.cc (hunk 4 above)
Comment (ZwX1616) [reply selfdrive/ui/qt/onroad.cc:280]: how else do you set it? scene is const?

File: selfdrive/ui/ui.h (hunk 1 above)
Comment (ZwX1616) [reply selfdrive/ui/ui.h:28]: yeah actually this was for something else, guess can remove

File: selfdrive/ui/ui.cc (hunk 2 above)
Comment (ZwX1616) [reply selfdrive/ui/ui.cc]: made a dm_rendered flag instead. This way the icon never shows if dm model packet for whatever reason doesn't show up

**Pull request merged**